Tests/SonarQube.MSBuild.Tasks.IntegrationTests/TargetsTests/BuildWrapperTargetsTests.cs

```diff
@@ -0,0 +1,182 @@
+        public void SQBeforeClCompile_DefaultBinPath_BinariesFound_TaskExecuted()
+        {
+            // Arrange
+            string rootInputFolder = TestUtils.CreateTestSpecificFolder(this.TestContext, "Inputs");
+            string rootOutputFolder = TestUtils.CreateTestSpecificFolder(this.TestContext, "Outputs");
+            BuildLogger logger = new BuildLogger();
+
+            WellKnownProjectProperties properties = new WellKnownProjectProperties();
+            properties.SonarQubeTempPath = rootInputFolder;
+            properties.BuildWrapperSkipFlag = "false";
+            properties.SonarQubeOutputPath = rootOutputFolder;
+
+            // By default, the targets try to find the build wrapper binaries in a specific folder
+            // below the folder containing the integration tasks assembly
+
+            ProjectRootElement projectRoot = BuildUtilities.CreateValidProjectRoot(this.TestContext, rootInputFolder, properties);
+
+            // Check we can find the integration targets file in the expected location
+            string targetsDir = TestUtils.GetTestSpecificFolderName(this.TestContext);
+            string integrationTargetsPath = Path.Combine(targetsDir, TestUtils.AnalysisTargetFile);
+            Assert.IsTrue(File.Exists(integrationTargetsPath), "Test setup error: did not find the integration targets in the expected location: {0}", integrationTargetsPath);
+
+            // The targets expect to find the build wrapper binaries in a folder under
+            // the folder containing the tasks assembly.
+            // Create a dummy tasks assembly
+            string dummyAsmFilePath = Path.Combine(targetsDir, "SonarQube.Integration.Tasks.dll");
+            Assert.IsFalse(File.Exists(dummyAsmFilePath), "Test setup error: not expecting the integration tasks assembly to exist at {0}", dummyAsmFilePath);
+            File.WriteAllText(dummyAsmFilePath, "dummy content");
+
+            // Create the build wrapper sub directory
+            string buildWrapperBinDir = Path.Combine(targetsDir, "build-wrapper-win-x86");
+            Directory.CreateDirectory(buildWrapperBinDir);
+
+            // Create the required build wrapper files relative to this location
+            CreateBuildWrapperMarkerFile(buildWrapperBinDir);
+
+            // Act
+            BuildUtilities.BuildTargets(projectRoot, logger, TargetConstants.BuildWrapperBeforeClCompileTarget);
+
+            // Assert
+            logger.AssertTargetExecuted(TargetConstants.BuildWrapperBeforeClCompileTarget);
+            logger.AssertTargetExecuted(TargetConstants.BuildWrapperAttachTarget);
+            // Note: task should fail because not all required files are present. However,
+            // the important thing for this test is that is executed.
+            logger.AssertTaskExecuted(TargetConstants.BuildWrapperAttachTask);
+        }
+
+        [TestMethod]
+        public void SQBeforeClCompile_SpecifiedWrapperLocation_BinariesFound_TaskExecuted()
```

Comment:
Checks that if the user specifies a location for the build wrapper binaries then it is used.